File: web-service/src/main/java/org/eclipse/model/Adresse.java
package org.eclipse.model;

import java.util.ArrayList;

public class Adresse {
	/** Classe pour les adresses */

	// Attributs
	private int id;
	private String numeroRue;
	private String nomRue;
	private String codePostal;
	private String ville;
	private String pays;
	private String complement;

	// Constructeurs
	public Adresse() {
		super();
	}

	public Adresse(int id) {
		super();
		this.id = id;
	}

	public Adresse(String numeroRue, String nomRue, String codePostal, String ville, String pays, String complement) {
		super();
		this.numeroRue = numeroRue;
		this.nomRue = nomRue;
		this.codePostal = codePostal;
		this.ville = ville;
		this.pays = pays;
		this.complement = complement;
	}

	public Adresse(int id, String numeroRue, String nomRue, String codePostal, String ville, String pays,
			String complement) {
		super();
		this.id = id;
		this.numeroRue = numeroRue;
		this.nomRue = nomRue;
		this.codePostal = codePostal;
		this.ville = ville;
		this.pays = pays;
		this.complement = complement;
	}

	// Getters et Setters
	public int getId() {
		return this.id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getNumeroRue() {
		return this.numeroRue;
	}

	public void setNumeroRue(String numeroRue) {
		this.numeroRue = numeroRue;
	}

	public String getNomRue() {
		return this.nomRue;
	}

	public void setNomRue(String nomRue) {
		this.nomRue = nomRue;
	}

	public String getCodePostal() {
		return this.codePostal;
	}

	public void setCodePostal(String codePostal) {
		this.codePostal = codePostal;
	}

	public String getVille() {
		return this.ville;
	}

	public void setVille(String ville) {
		this.ville = ville;
	}

	public String getPays() {
		return this.pays;
	}

	public void setPays(String pays) {
		this.pays = pays;
	}

	public String getComplement() {
		return this.complement;
	}

	public void setComplement(String complement) {
		this.complement = complement;
	}

	// toString
	@Override
	public String toString() {
		return "Adresse [id=" + this.id + ", numeroRue=" + this.numeroRue + ", nomRue=" + this.nomRue + ", codePostal="
				+ this.codePostal + ", ville=" + this.ville + ", pays=" + this.pays + ", complement=" + this.complement
				+ "]";
	}

}

File: angular-server/src/app/classes/model/adresse.ts
export class Adresse {
    constructor(private _id: bigint = BigInt(0), private _numeroRue: string, private _nomRue: string, private _codePostal: string, private _ville: string, private _pays: string | null , private _complement: string | null) { }

    get id(): bigint {
        return this._id;
    }

    set id(_id: bigint) {
        this._id = _id;
    }

    get numeroRue(): string {
        return this._numeroRue;
    }

    set numeroRue(_numeroRue : string) {
        this._numeroRue = _numeroRue;
    }

    get nomRue(): string  {
        return this._nomRue;
    }

    set nomRue(_nomRue : string) {
        this._nomRue = _nomRue;
    } 

    get codePostal(): string {
        return this._codePostal;
    }

    set codePostal(_codePostal: string) {
        this._codePostal = _codePostal;
    }

    get ville(): string {
        return this._ville;
    }

    set ville(_ville: string) {
        this._ville = _ville;
    }

    get pays(): string | null {
        return this._pays;
    }

    set pays(_pays: string |null) {
        this._pays = _pays;
    }

    get complement(): string | null {
        return this._complement;
    }

    set complement(_complement: string |null) {
        this._complement = _complement;
    }

}
